src/library_webgl.js

```diff
@@ -738,60 +788,6 @@ for (/**@suppress{duplicate}*/var i = 0; i < {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
       }
       disableHalfFloatExtensionIfBroken(ctx);
 #endif
-
-      // If end user enables *glGetProcAddress() functionality or
-      // GL_SUPPORT_AUTOMATIC_ENABLE_EXTENSIONS, then we must filter out
-      // all future WebGL extensions from being passed to the user, and only restrict to advertising
-      // extensions that the *glGetProcAddress() function knows to handle.
-#if GL_ENABLE_GET_PROC_ADDRESS || GL_SUPPORT_AUTOMATIC_ENABLE_EXTENSIONS
-      var _allSupportedExtensions = ctx.getSupportedExtensions;
-      var supportedExtensionsForGetProcAddress = [
-#if MIN_WEBGL_VERSION == 1
-        // WebGL 1 extensions
-        'ANGLE_instanced_arrays',
-        'EXT_blend_minmax',
-        'EXT_disjoint_timer_query',
-        'EXT_frag_depth',
-        'EXT_shader_texture_lod',
-        'EXT_sRGB',
-        'OES_element_index_uint',
-        'OES_fbo_render_mipmap',
-        'OES_standard_derivatives',
-        'OES_texture_float',
-        'OES_texture_half_float',
-        'OES_texture_half_float_linear',
-        'OES_vertex_array_object',
-        'WEBGL_color_buffer_float',
-        'WEBGL_depth_texture',
-        'WEBGL_draw_buffers',
-#endif
-#if MAX_WEBGL_VERSION >= 2
-        // WebGL 2 extensions
-        'EXT_color_buffer_float',
-        'EXT_disjoint_timer_query_webgl2',
-        'EXT_texture_norm16',
-        'WEBGL_clip_cull_distance',
-#endif
-        // WebGL 1 and WebGL 2 extensions
-        'EXT_color_buffer_half_float',
-        'EXT_float_blend',
-        'EXT_texture_compression_bptc',
-        'EXT_texture_compression_rgtc',
-        'EXT_texture_filter_anisotropic',
-        'KHR_parallel_shader_compile',
-        'OES_texture_float_linear',
-        'WEBGL_compressed_texture_s3tc',
-        'WEBGL_compressed_texture_s3tc_srgb',
-        'WEBGL_debug_renderer_info',
-        'WEBGL_debug_shaders',
-        'WEBGL_lose_context',
-        'WEBGL_multi_draw',
-      ];
-      ctx.getSupportedExtensions = function() {
```

Comment:
I wonder if there is any risk to removing this shim - now any user JS code that checks the list of supported extensions (on the context directly) will get a different list than WASM code. I think it is OK though.

Reply:
I don't think user code should be doing anything directly with the GL context though should it?  If it does it certainly should know what its doing.

Reply:
I think it's not uncommon to use the context inside EM_ASM blocks, for example to use interop entrypoints like `texImage2D(ImageBitmap)`. There are probably other cases of applications mixing JS and WASM code for various reasons, and either grabbing the context out of Emscripten or calling `canvas.getContext('webgl')` which will re-return the same context object. But it wouldn't be so common for such code to be enabling extensions.

Also I forgot this shim is new, and 2 weeks ago the JS-exposed extension list was never filtered, so I don't think there's risk of breaking existing applications. Only risk of confusing developers slightly, I think.

Reply:
I also think there is an argument for not filter the list for users that directly go under the hood and access the context.  In that case, such users are more likely to want to get the unfiltered list, since in those circumstances they might not be limited to extensions that emscripten's C++ library supports and they might want to see the real/underlying list